iosApp/iosApp/ContentView.swift

```diff
@@ -167,7 +167,18 @@ struct ContentView: View {
                         if !searchObserver.isSearching, !viewportProvider.viewport.isFollowing,
                            locationDataManager.currentLocation != nil {
                             VStack(alignment: .trailing) {
-                                RecenterButton { Task { viewportProvider.follow() } }
+                                RecenterButton(icon: .faLocationArrowSolid, size: 17.33) {
+                                    viewportProvider.follow()
+                                }
+                                .accessibilityIdentifier("mapRecenterButton")
+                            }.frame(maxWidth: .infinity, alignment: .topTrailing)
+                        }
+                        if !searchObserver.isSearching, !viewportProvider.viewport.isOverview,
+                           let (routeType, selectedVehicle, stop) = recenterOnVehicleButtonInfo() {
```

Comment:
This causes some weird behavior when you tap between single route+direction stops on the map, when the viewport is moving to the new stop overview, the vehicle recenter button flickers in for a second. Not certain why, but I think it might be when the old selected vehicle is still set but the viewport is set to the position of the new stop, before the auto trip filter has loaded and set the new vehicle, which sets the viewport to overview once it changes. You might be able to include a check that `selectedVehicle.id == tripFilter.vehicleId`?

Reply:
That appears to work. Good catch.